### PR TITLE
Editor done button and delete node - handles #53 #54 #61 #57

### LINKED ISSUE #53: Implement Delete Node
Do from node editor button and right click menu.

### PR DESCRIPTION
**Pull request by sachingrover211**
Resolves #53, #54, #61, #57.

### CODE REVIEW

**Comment by sachingrover211** [comment]
Master merged to the branch. Please do the code review for all the above issues.

**Comment by sachingrover211** [comment]
@jonwetzel @riteshsamala reminder for the pull request.

File: www/js/model.js [hunk 1]
```diff
@@ -329,6 +329,28 @@ define([
...
+			isComplete: function(/*string*/ id){
+				var node = this.getNode(id);
+				// if units were not entered even then it would show node complete
+				var unitsOptional = true;
+				var returnFlag = '';
+
+				var nameEntered = node.type && node.type == "equation" || node.variable != null;
+				var valueEntered = node.type && node.type == "equation" || node.value != null;
+				var equationEntered = node.type && node.type == "quantity" || node.value != null;
+				if(node.genus == "required" || node.genus == "allowed" || node.genus == "preferred"){
```
Comment (riteshsamala) [inline www/js/model.js]: genus == "preferred" no longer used !

**Comment by sachingrover211** [comment]
@riteshsamala all the review comments have been incorporated. Please check. there were two commits for these.

**Pull request merged**